src/main/java/br/com/glad/gladtask/controllers/UserController.java

```diff
@@ -58,12 +57,18 @@ public UserController(UserService userService, TeamService teamService, Password
 
 			final User savedUser = createOrUpdateUser(userFromRequest);
 
-			String teamId = json.get("teamId").asText();
+			String teamId = json.get("teamId") != null ? json.get("teamId").asText() : null;
 			if (teamId != null) {
```

Comment:
just to make it simple, you can validate with is null or not with the json.get("teamId") and just after that, get the value and pass to teamId. Because, the way it is, you are doing twice if/else... for the same thing.

Reply:
Adjusted